### PR TITLE
[MIG][10.0] account_analytic_distribution_required module

### PR DESCRIPTION
**Pull request by cubells**
Depends on
- [x] #100

Migration of module account_analytic_plan_required to version 10.0

- Rename module
- Updated README
- Updated tests

cc @Tecnativa

### CODE REVIEW

File: account_analytic_distribution_required/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/__init__.py]: Remove copyright here.

File: account_analytic_distribution_required/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/__manifest__.py]: Add our copyright

File: account_analytic_distribution_required/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Account Analytic Plan Required',
+    'version': '10.0.0.1.0',
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/__manifest__.py]: 10.0.1.0.0 (you always switch this one)

File: account_analytic_distribution_required/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    'category': 'Analytic Accounting',
+    'license': 'AGPL-3',
+    'author': "ACSONE SA/NV, "
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/__manifest__.py]: Add author

File: account_analytic_distribution_required/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/models/__init__.py]: Remove copyright here

File: account_analytic_distribution_required/models/account.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/models/account.py]: Add our copyright

File: account_analytic_distribution_required/models/account.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                return _("Analytic policy is set to "
+                         "'Always (analytic account or distribution)' "
+                         "with account %s '%s' but the analytic "
+                         "distribution and the analytic account are "
+                         "missing in the account move line "
+                         "with label '%s'.") % \
+                        (move_line.account_id.code,
+                         move_line.account_id.name,
+                         move_line.name)
+            elif (analytic_policy == 'always' and
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/models/account.py]: You should call here super in other cases different from the new two options.

File: account_analytic_distribution_required/models/account.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+                         ) % (move_line.account_id.code,
+                              move_line.account_id.name,
+                              move_line.name)
+
+    @api.constrains('analytic_account_id', 'analytic_distribution_id',
+                    'account_id', 'debit', 'credit')
+    def _check_analytic_required(self):
+        for rec in self:
+            message = rec._check_analytic_distribution_required_msg()
+            if message:
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/models/account.py]: Or better call here `super` if there's no message

File: account_analytic_distribution_required/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/tests/__init__.py]: Remove copyright here

File: account_analytic_distribution_required/tests/test_account_analytic_plan_required.py [hunk 10]
```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Acsone - Stéphane Bidoul <stephane.bidoul@acsone.eu>
```
Comment (pedrobaeza) [inline account_analytic_distribution_required/tests/test_account_analytic_plan_required.py]: Add our copyright

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
download.gna.org seems to be down and this makes Travis to fail. Let's wait on that.

**Comment by pedrobaeza** [comment]
Is also included in OpenUpgrade the module renaming?

**Comment by cubells** [comment]
@pedrobaeza 

yes, it is: https://github.com/OCA/OpenUpgrade/pull/842

**Review by pedrobaeza — APPROVED** [review]
Tentatively approved until #100 is merged

**Comment by pedrobaeza** [comment]
@luismontalba please review

**Review by luismontalba — APPROVED** [review]
Tested on runbot, it works

**Pull request merged**